### PR TITLE
Use in-memory pull-secret instead of writing to disk

### PR DESCRIPTION
**Pull request by pmtk**
Using pull secret in-memory makes running the operator easier in out-of-cluster scenario (using `go run .` against `$KUBECONFIG`)

### CODE REVIEW

File: pkg/registry/registry.go [hunk 1]
```diff
@@ -60,32 +60,67 @@ type registry struct {
 	log        logr.Logger
 }
 
-func (r *registry) writeImageRegistryCredentials(ctx context.Context) error {
+type dockerAuth struct {
+	Auth  string
+	Email string
+}
+
+func (r *registry) registryFromImageURL(image string) (string, error) {
```
Comment (qbarrand) [inline pkg/registry/registry.go:68]: Could we have a unit test for this function?

File: pkg/registry/registry.go [hunk 2]
```diff
@@ -60,32 +60,67 @@ type registry struct {
...
+	}{}
+
+	err = json.Unmarshal(pullSecretData, &auths)
 	if err != nil {
-		return errors.Wrap(err, "Can not write pull secret file")
+		return dockerAuth{}, errors.New("Failed to unmarshal auths")
+	}
+
+	if auth, ok := auths.Auths[registry]; !ok {
+		return dockerAuth{}, errors.New("Cluster PullSecret does not contain auth for the registry " + registry)
```
Comment (qbarrand) [inline pkg/registry/registry.go:110]:
```suggestion
		return dockerAuth{}, fmt.Errorf("cluster PullSecret does not contain auth for registry %s", registry)
```

File: pkg/registry/registry.go [hunk 3]
```diff
@@ -60,32 +60,67 @@ type registry struct {
...
 	s, err := r.kubeClient.GetSecret(ctx, pullSecretNamespace, pullSecretName, metav1.GetOptions{})
 	if err != nil {
-		return errors.Wrap(err, "Can not retrieve pull secrets")
+		return dockerAuth{}, errors.Wrap(err, "Can not retrieve pull secrets")
 	}
 
 	pullSecretData, ok := s.Data[pullSecretFileName]
 	if !ok {
-		return errors.New("Can not find data content in the secret")
+		return dockerAuth{}, errors.New("Can not find data content in the secret")
```
Comment (qbarrand) [inline pkg/registry/registry.go:97]:
```suggestion
		return dockerAuth{}, errors.New("could not find data content in the secret")
```

**Review by pacevedom — COMMENTED** [review]
I am missing the changes to drop the volume in the manifests.

File: pkg/registry/registry.go (hunk 1 above)
Comment (pmtk) [reply pkg/registry/registry.go:68]: Yes, added :)

File: pkg/registry/registry.go [hunk 4]
```diff
@@ -60,32 +60,67 @@ type registry struct {
...
+	return u.Host, nil
+}
+
+func (r *registry) getImageRegistryCredentials(ctx context.Context, registry string) (dockerAuth, error) {
 	_, err := r.kubeClient.GetNamespace(ctx, pullSecretNamespace, metav1.GetOptions{})
 	if err != nil {
 		r.log.Info("Can not find namespace for pull secrets, assuming vanilla k8s")
-		return nil
+		return dockerAuth{}, nil
 	}
```
[the remark this answers is not in the document]
Comment (pmtk) [reply pkg/registry/registry.go]: Removed

**Pull request merged**